src/rgw/rgw_rados.cc

```diff
@@ -8408,8 +8408,13 @@ int RGWRados::delete_system_obj(rgw_raw_obj& obj, RGWObjVersionTracker *objv_tra
   return 0;
 }
 
-int RGWRados::delete_obj_index(const rgw_obj& obj)
+int RGWRados::delete_obj_index(const rgw_obj& _obj)
 {
+  rgw_obj obj = _obj;
+  if (obj.key.instance == "null") {
```

Comment:
When we handle the obj index, we should clear the instance if it's "null". Like delete_obj does:
int RGWRados::Object::Delete::delete_obj()
{
  RGWRados *store = target->get_store();
  rgw_obj& src_obj = target->get_obj();
  const string& instance = src_obj.get_instance();
  rgw_obj obj = src_obj;

  if (instance == "null") {
    obj.clear_instance();
  }